src/chainstate/burn/db/sortdb.rs

```diff
@@ -992,6 +1017,92 @@ impl <'a> SortitionHandleConn <'a> {
         Ok(key_status)
     }
 
+    /// Return a vec of sortition winner's burn header hash and stacks header hash, ordered by
+    ///   increasing block height in the range (block_height_begin, block_height_end]
+    fn get_sortition_winners_in_fork(&self, block_height_begin: u32, block_height_end: u32) -> Result<Vec<(Txid, u64)>,  BurnchainError> {
+        let mut result = vec![];
+        for height in (block_height_begin+1)..(block_height_end+1) {
+            debug!("Looking for winners at height = {}", height);
+            let snapshot = SortitionDB::get_ancestor_snapshot(self, height as u64, &self.context.chain_tip)?
+                .ok_or_else(|| BurnchainError::MissingParentBlock)?;
+            if snapshot.sortition {
+                result.push((snapshot.winning_block_txid, snapshot.block_height));
+            }
+        }
+        Ok(result)
+    }
+
+    pub fn get_reward_cycle_info(&self, prepare_end: &BurnchainHeaderHash, pox_consts: &PoxConstants) -> Result<Option<(ConsensusHash, BlockHeaderHash)>, CoordinatorError> {
+        let prepare_end_sortid = self.get_sortition_id_for_bhh(prepare_end)?
+            .ok_or_else(|| BurnchainError::MissingParentBlock)?;
+        let my_height = SortitionDB::get_block_height(self.deref(), &prepare_end_sortid)?
+            .expect("CORRUPTION: SortitionID known, but no block height in SQL store");
+
+        // there can be either 1 or 0 PoX anchors.
+        assert!(pox_consts.anchor_threshold > (pox_consts.prepare_length / 2));
+
+        // if this block is the _end_ of a prepare phase,
+        if ! (my_height % pox_consts.reward_cycle_length == 0) {
```

Comment:
I think it's 2999.  The reward cycle is blocks 2760-2999, which spans 240 blocks.

Reply:
Unrelated, should there be an additional check here to make sure `my_height` falls within a registration period?

Reply:
just to be precise, @jcnelson you meant `The prepare cycle is blocks 2760-2999`?

Reply:
The way it works right now is that the prepare phase would be blocks 2761-3000 -- so that the last block of the prepare phase is `0 mod reward_cycle_length`.

> Unrelated, should there be an additional check here to make sure my_height falls within a registration period?

I'm not sure -- the preceding check ensures that this block is exactly the last block of a prepare phase.